pkg/reconciler/kubernetes/tektonresult/transform.go

```diff
@@ -0,0 +1,217 @@
+		common.ApplyProxySettings,
+		common.ReplaceNamespaceInDeploymentArgs(targetNs),
+		common.ReplaceNamespaceInDeploymentEnv(targetNs),
+		updateApiConfig(ctx, instance.Spec.ResultsAPIProperties),
+		enablePVCLogging(ctx, instance.Spec.ResultsAPIProperties),
+		common.AddDeploymentRestrictedPSA(),
+		common.AddStatefulSetRestrictedPSA(),
+	}
+	extra = append(extra, r.extension.Transformers(instance)...)
+	return common.Transform(ctx, manifest, instance, extra...)
+}
+
+// ReplaceNamespaceInDeploymentArgs replaces namespace in deployment's args
+func enablePVCLogging(ctx context.Context, p v1alpha1.ResultsAPIProperties) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "Deployment" || u.GetName() != DeploymentAPI || p.LOGGING_PVC_NAME == "" {
+			return nil
+		}
+
+		logger := logging.FromContext(ctx)
+		d := &appsv1.Deployment{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, d)
+		if err != nil {
+			return err
+		}
+
+		set := true
+		for i := 0; i < len(d.Spec.Template.Spec.Containers[0].VolumeMounts); i++ {
+			if d.Spec.Template.Spec.Containers[0].VolumeMounts[i].Name == pvcLoggingVolume {
+				d.Spec.Template.Spec.Containers[0].VolumeMounts[i] = corev1.VolumeMount{Name: pvcLoggingVolume,
+					MountPath: p.LOGS_PATH}
+				logger.Debugf("Volumemounts set for tekton results logging")
+				set = false
+			}
+		}
+		if set {
+			d.Spec.Template.Spec.Containers[0].VolumeMounts = append(d.Spec.Template.Spec.Containers[0].VolumeMounts,
+				corev1.VolumeMount{Name: pvcLoggingVolume,
+					MountPath: p.LOGS_PATH})
+		}
+
+		set = true
+		vol := corev1.Volume{
+			Name: pvcLoggingVolume,
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: p.LOGGING_PVC_NAME,
+				},
+			},
+		}
+		for i := 0; i < len(d.Spec.Template.Spec.Volumes); i++ {
+			if d.Spec.Template.Spec.Volumes[0].Name == pvcLoggingVolume {
+				d.Spec.Template.Spec.Volumes[0] = vol
+				logger.Infof("PVC Volume set for tekton results")
+				set = false
+			}
+		}
+		if set {
+			d.Spec.Template.Spec.Volumes = append(d.Spec.Template.Spec.Volumes, vol)
+		}
+
+		unstrObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(d)
+		if err != nil {
+			return err
+		}
+		u.SetUnstructuredContent(unstrObj)
+
+		return nil
+	}
+}
+
+func updateApiConfig(ctx context.Context, p interface{}) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+
+		kind := strings.ToLower(u.GetKind())
+		if kind != "configmap" {
+			return nil
+		}
+
+		if u.GetName() != ConfigAPI {
+			return nil
+		}
+
+		logger := logging.FromContext(ctx)
+
+		cm := &corev1.ConfigMap{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, cm)
+		if err != nil {
+			return err
+		}
+		if cm.Data == nil {
+			cm.Data = map[string]string{}
+		}
+
+		values := reflect.ValueOf(p)
```

Comment:
We can't convert struct to map without using reflect.

Reply:
I had to use string transformation in between. Configmap and Json have a mismatch. Also, I saw some transformers not using that but that package isn't that helpful in our case because we do customize transformations.